web/src/app/services/temp-survey/temp-survey.service.ts

```diff
@@ -0,0 +1,109 @@
+import {Injectable} from '@angular/core';
+import {Survey} from 'app/models/survey.model';
+import {DataStoreService} from '../data-store/data-store.service';
+import {BehaviorSubject, Observable, firstValueFrom} from 'rxjs';
+import {LocationOfInterest} from 'app/models/loi.model';
+import {List} from 'immutable';
+import {Job} from 'app/models/job.model';
+import { Task } from 'app/models/task/task.model';
+import { LocationOfInterestService } from '../loi/loi.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TempSurveyService {
+  private tempSurvey$$!: BehaviorSubject<Survey>;
+  private tempLois$$!: BehaviorSubject<List<LocationOfInterest>>;
+
+  private tempSurvey!: Survey;
+  private tempLois!: List<LocationOfInterest>;
+
+  constructor(private dataStore: DataStoreService) {}
+
+  async init(id: string) {
+    this.tempSurvey = await firstValueFrom(this.dataStore.loadSurvey$(id));
+    this.tempSurvey$$ = new BehaviorSubject<Survey>(this.tempSurvey);
+
+    this.tempLois = await firstValueFrom(this.dataStore.lois$({id} as Survey));
```

Comment:
We don't need to attempt solve the LOIs problem yet. Import is done server side, so creating an "unsaved draft" state would imply managing a separate collection in Firestore for draft LOIs. That's probably out of scope for this chang.

Reply:
yes, I wanted to keep it separated. if we don't have to care about it now it makes this pr easier